### PR TITLE
Integrate DrugBAN example to PyKale

### PR DESCRIPTION
**Pull request by jiayang-zhang**
Fixes #{issue_number}.

### Description
Integrated Peizhen's DrugBAN to PyKale's workflow. 
 
1. Added a new folder for DrugBAN in the `examples` directory as a starting point for using this model within PyKale.
2. Added `drugban.py` and `ban.py` to `kale.embed` for DrugBAN and BANLayer.
3. Added `drugban_datasets.py` to `kale.loaddata` for loading datasets specific to DrugBAN.
4. Added `drugban_trainer` to kale.pipeline.
5. Created a `Discriminator` class in `kale.predict.class_domain_nets.py` to support domain adaptation.
6. Added `binary_cross_entropy` and `drugban_cross_entropy_logits` functions in `kale.predict.losses` for loss calculation.
7. Added `drugban_utils` in `kale.utils` for helper functions.

### Status
**Work in progress**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] In-line docstrings updated.
- [ ] [Source for documentation at `docs`](https://github.com/pykale/pykale/tree/main/docs/source) manually updated for new API.

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -17,7 +17,8 @@ All data in examples are from the public domain so they will be downloaded into
   - [Action recognition via domain adaptation](https://github.com/pykale/pykale/tree/master/examples/action_dann)
   - [Video loading](https://github.com/pykale/pykale/tree/master/examples/video_loading)
 - Bioinformatics data analysis
-  - [Drug-target interaction prediction](https://github.com/pykale/pykale/tree/master/examples/bindingdb_deepdta)
+  - [Drug-target interaction prediction (with Deepdta)](https://github.com/pykale/pykale/tree/master/examples/bindingdb_deepdta)
```
Comment (jiayang-zhang) [inline examples/README.md:20]: Change to "DeepDTA"
